ported/librdmacm/cmatose.c

```diff
@@ -379,7 +383,9 @@ static int shutdown_events(void)
 
 	while (disconnects_left && !ret) {
 		ret = fi_eq_sread(eq, &event, &entry, sizeof entry, -1, 0);
-		if (ret < 0) {
+		if (ret == -FI_EAGAIN)
+			continue;
```

Comment:
Actually in both loops above, we're blocking forever until we read an event.  So even seeing EAGAIN would be unexpected.  Both changes above should probably be reverted.